apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts

```diff
@@ -183,7 +173,37 @@ export class ParseEventRequest {
     // eslint-disable-next-line no-param-reassign
     command.payload = merge({}, defaultPayload, command.payload);
 
-    return await this.dispatchEvent(command, transactionId);
+    const result = await this.dispatchEvent(command, transactionId);
+
+    if (!template.connected && !command.payload?.__source) {
+      await this.notificationTemplateRepository.updateOne(
+        {
+          _environmentId: command.environmentId,
+          _id: template._id,
+        },
+        {
+          $set: {
+            connected: true,
```

Comment:
Open to any suggestions for a better name here, just followed the same convention from Integraitons

Reply:
I didn't wanted to go with lastTriggered because this might cause a lot of MongoDB traffic on high throughput workflows, Altho I do agree that seeing a last triggered is very useful in the dashboard. Any ideas on how we can limit the update behaviour to avoid 1 to 1 mapping of triggers to update calls on the db table? 

We do use caching for the template during the workflow execution, so document lock might not be a huge concern. 🤔

Reply:
I suggest updating it once every minute. A solution would be:

1. Update `getNotificationTemplateByTriggerIdentifier` in `parse-event-request` to have a ttl of 1 minute. 
2. Replace the findByTriggerIdentifier with a findOneAndUpdate that when invoked it updates the `lastRunAt`.

That way, every time parse-event-request is executed, it will either get the cached value for a minute, or the updated value from Mongo that will be cached for another minute.